Added CashRegister.cs class

CCianfloneDev opened the pull request.

CCianfloneDev committed:
Add files via upload

CCianfloneDev committed:
Add files via upload

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+    public class CashRegister
+    {
+        private decimal creditBalance;
+        private decimal debitBalance;
```
- [x] Add `cashBalance` to distinguish between cash, and debit card purchases

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+        private int storeId;
+
+        /// <summary>
+        /// Gets the credit balance of a cash register.
```
- [x] Be sure to clarify if the credit balance will be negative or positive.

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+                return creditBalance;
+            }
+
+            private set
+            {
+                creditBalance = value;
+            }
```
- [x] Add exception checking.

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+                return debitBalance;
+            }
+
+            private set
+            {
+                debitBalance = value;
+            }
```
- [x] Add exception checking.

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+            }
+        }
+
+        /// <summary>
+        /// Gets sales tax rate.
+        /// </summary>
+        public decimal SalesTaxRate
+        {
+            get
+            {
+                return salesTaxRate;
+            }
+            private set
+            {
+                salesTaxRate = value;
+            }
+        }
```
- [x] Split it into separated taxes for GST & PST

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+                salesTaxRate = value;
+            }
+        }
+        /// <summary>
+        /// Gets the store ID.
+        /// </summary>
+        public int StoreId
+        {
+            get
+            {
+                return storeId;
+            }
+
+            private set
+            {
+                storeId = value;
+            }
+        }
```
- [x] Reference the store instead of the ID. It's easier to write the exception for making sure you have a valid store.
- [x] Add exceptions for checking to ensure you aren't referencing an invalid store.

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+        /// </summary>
+        /// <param name="creditBalance">Credit balance.</param>
+        /// <param name="debitBalance">Debit balance.</param>
+        /// <param name="storeId">Store ID.</param>
```
- [x] It is better to construct with a reference to the store than with a `storeID`

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+        /// <param name="storeId">Store ID.</param>
+        public CashRegister(decimal creditBalance, decimal debitBalance, int storeId)
+        {
+            CreditBalance = creditBalance;
+            DebitBalance = debitBalance;
+            StoreId = storeId;
```
- [x] Add exception checking.

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+        /// Processes payment with given fact that it is debit/cash or credit.
+        /// </summary>
+        /// <param name="isDebitOrCash">The fact that it is debit/cash or credit.</param>
+        public void ProcessPayment(bool isDebitOrCash)
```
- [x] May need a new enum for `PaymentType`

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+        public void ProcessPayment(bool isDebitOrCash)
+        {
+            decimal totalCostOfPurchase = 0;
+            List<int> listOfItemPricesAtPurchase = new List<int>();
```
- [x] Prices should be `decimal`s

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+        public void ProcessPayment(bool isDebitOrCash)
+        {
+            decimal totalCostOfPurchase = 0;
+            List<int> listOfItemPricesAtPurchase = new List<int>();
+            List<ItemId> listOfItems = new List<ItemId>();
```
- [x] Is there a data source where these lists should be coming from?

DZ-FSDev reviewed (changes requested):
**CR for #24**

CCianfloneDev committed:
Add files via upload

CCianfloneDev committed:
Update ItemId.cs

added more items + documentation.

CCianfloneDev committed:
Merge pull request #25 from CCianfloneDev/CCianfloneDev-ItemId-enum

Added  ItemId.cs enum

DZ-FSDev commented on ColesStopAndShop/CashRegister.cs:
```diff
@@ -0,0 +1,178 @@
+
+            Console.WriteLine("Items: ");
+
+            for (int i = 0; i < listOfItems.Count; i++)
+            {
+                for (int j = 0; j < listOfItems.Count; j++)
+                {
+                    Console.WriteLine($" {listOfItemPricesAtPurchase[i]} :  {itemsBoughtToString[i]}");
+                }
+            }
```
- [x] Refactor to a single loop. The 2D loop here is not required and is causing #27.

DZ-FSDev reviewed (changes requested):
**Added CR for #24 Ref: #27**

CCianfloneDev committed:
Merge pull request #28 from CCianfloneDev/CCianfloneDev-program-class

Added program.cs

CCianfloneDev committed:
Update CashRegister.cs

Added cashBalance field.

CCianfloneDev committed:
Update CashRegister.cs

Corrected documentation for creditBalance.

CCianfloneDev committed:
Update CashRegister.cs

Added exception checking for CreditBalance.

CCianfloneDev committed:
Update CashRegister.cs

Added exception checking for DebitBalance.

CCianfloneDev committed:
Update CashRegister.cs

Added GST and PST rate. Kept SalesTaxRate prop but it now combines PST and GST for easier total calculation within PaymentProcess.

CCianfloneDev committed:
Update CashRegister.cs

Removed association to store via a storeId. It now is associated to a store via a reference to a constructed GasStation.

CCianfloneDev committed:
Update CashRegister.cs

Added exception checking within constructor.

CCianfloneDev committed:
Update CashRegister.cs

Changed prices from int to decimal

CCianfloneDev committed:
Created PaymentType.cs enum

CCianfloneDev committed:
Update CashRegister.cs

Corrected logic problem with 2d loop. 
Cash register no longer requires cashier to enter in price of item, the cashier enters in itemId (mimics scanning barcode) and it is used to reference the price of that item which is defined in a dict within the gas station.

CCianfloneDev committed:
Update GasStation.cs

CCianfloneDev committed:
Update Program.cs

DZ-FSDev reviewed (approved):
Good work!
